### PR TITLE
Add settings for auto-renewal of Edge CA

### PR DESCRIPTION
**Pull request by gordonwang0**
Updates settings, tests, and config apply tool to add support for the auto-renewal of the Edge CA. The actual auto-renewal code will come in a later PR.

### CODE REVIEW

File: edgelet/contrib/config/linux/template.toml [hunk 1]
```diff
@@ -387,6 +387,24 @@
...
 # auto_generated_edge_ca_expiry_days = 90
 
+# ==============================================================================
+# Edge CA certificate auto-renewal
+# ==============================================================================
+#
+# Manage auto-renewal of the Edge CA certificate. Generally, this applies when the
+# Edge CA is configured as Quickstart or when the Edge CA has an issuance `method`
+# set. Edge CA certs loaded from files generally cannot be auto-renewed as the Edge
+# runtime will not have enough information to renew them.
```
Comment (varunpuranik) [inline edgelet/contrib/config/linux/template.toml:397]: In case of Edge CA loaded from file, I suppose we will log a warning that we cannot renew cert and continue operations, right?

File: edgelet/iotedge/test-files/config/ca-quickstart/keyd.toml [hunk 2]
```diff
@@ -11,10 +11,10 @@ device-id = "file:///var/secrets/aziot/keyd/device-id"
 uid = 5556
 keys = ["aziot_identityd_master_id", "device-id"]
 
-[[principal]]
```
Comment (varunpuranik) [inline edgelet/iotedge/test-files/config/ca-quickstart/keyd.toml:14]: why this change in this file, and the files below?

File: edgelet/contrib/config/linux/template.toml (hunk 1 above)
Comment (gordonwang0) [reply edgelet/contrib/config/linux/template.toml:397]: No, I left out the setting for preloaded certs. It will have no effect if set.

File: edgelet/iotedge/test-files/config/ca-quickstart/keyd.toml (hunk 2 above)
Comment (gordonwang0) [reply edgelet/iotedge/test-files/config/ca-quickstart/keyd.toml:14]: In order to add "aziot-edged-ca-temp" to all files cleanly, I had to rearrange some of the output ordering. That causes this principal to come before the previous. Ordering of these principals has no effect at runtime though.

**Review by arsing — APPROVED** [review]
